packages/@aws-cdk/aws-codebuild/lib/build-spec.ts

```diff
@@ -0,0 +1,101 @@
+import { IResolveContext, Lazy, Stack } from '@aws-cdk/cdk';
+
+/**
+ * BuildSpec for CodeBuild projects
+ */
+export abstract class BuildSpec {
+  public static fromObject(value: {[key: string]: any}): BuildSpec {
+    return new StructuredBuildSpec(value);
+  }
+
+  /**
+   * Use a file from the source as buildspec
+   *
+   * Use this if you want to use a file different from 'buildspec.yml'`
+   */
+  public static fromFilename(filename: string): BuildSpec {
+    return new FilenameBuildSpec(filename);
+  }
+
+  protected constructor() {
+  }
+
+  /**
+   * Render the represented BuildSpec
+   */
+  public abstract toBuildSpec(): string;
+}
+
+/**
+ * BuildSpec that just returns the input unchanged
+ */
+class FilenameBuildSpec extends BuildSpec {
+  constructor(private readonly filename: string) {
+    super();
+  }
+
+  public toBuildSpec(): string {
+    return this.filename;
+  }
+
+  public toString() {
+    return `<buildspec file: ${this.filename}>`;
+  }
+}
+
+/**
+ * BuildSpec that understands about structure
+ */
+class StructuredBuildSpec extends BuildSpec {
```

Comment:
This name is surprising to me... I would actually say this is the complete opposite of a `StructuredBuildSpec` - it's an `UnstructuredBuildSpec`, because I can say `new StructuredBuildSpec({ foo: 123 })`, and nothing would complain (until deployment, that is...).

If `UnstructuredBuildSpec` is too weird, maybe `ObjectBuildSpec`?